src/garbage_collector.rs

```diff
@@ -0,0 +1,208 @@
+    main_thread: Option<JoinHandle<()>>,
+}
+
+impl GarbageCollector {
+    pub(crate) async fn new(
+        manifest_store: Arc<ManifestStore>,
+        table_store: Arc<TableStore>,
+        options: GarbageCollectorOptions,
+        tokio_handle: Handle,
+        db_stats: Arc<DbStats>,
+    ) -> Result<Self, SlateDBError> {
+        let (external_tx, external_rx) = crossbeam_channel::unbounded();
+        let (err_tx, err_rx) = tokio::sync::oneshot::channel();
+        let tokio_handle = options.compaction_runtime.clone().unwrap_or(tokio_handle);
+        let main_thread = thread::spawn(move || {
+            let load_result = GarbageCollectorOrchestrator::new(
+                manifest_store,
+                table_store,
+                options,
+                tokio_handle,
+                external_rx,
+                db_stats,
+            );
+            let mut orchestrator = match load_result {
+                Ok(orchestrator) => orchestrator,
+                Err(err) => {
+                    err_tx.send(Err(err)).expect("err channel failure");
+                    return;
+                }
+            };
+            err_tx.send(Ok(())).expect("err channel failure");
+            orchestrator.run();
+        });
+        err_rx.await.expect("err channel failure")?;
+        Ok(Self {
+            main_thread: Some(main_thread),
+            main_tx: external_tx,
+        })
+    }
+
+    /// Close the garbage collector
+    pub(crate) async fn close(mut self) {
+        if let Some(main_thread) = self.main_thread.take() {
+            self.main_tx.send(Shutdown).expect("main tx disconnected");
+            main_thread
+                .join()
+                .expect("failed to stop main compactor thread");
+        }
+    }
+}
+
+struct GarbageCollectorOrchestrator {
+    manifest_store: Arc<ManifestStore>,
+    table_store: Arc<TableStore>,
+    options: GarbageCollectorOptions,
+    stored_manifest: StoredManifest,
+    tokio_handle: Handle,
+    external_rx: crossbeam_channel::Receiver<GarbageCollectorMessage>,
+    db_stats: Arc<DbStats>,
+}
+
+impl GarbageCollectorOrchestrator {
+    fn new(
+        manifest_store: Arc<ManifestStore>,
+        table_store: Arc<TableStore>,
+        options: GarbageCollectorOptions,
+        tokio_handle: Handle,
+        external_rx: crossbeam_channel::Receiver<GarbageCollectorMessage>,
+        db_stats: Arc<DbStats>,
+    ) -> Result<Self, SlateDBError> {
+        let stored_manifest =
+            tokio_handle.block_on(StoredManifest::load(manifest_store.clone()))?;
+        let Some(stored_manifest) = stored_manifest else {
+            return Err(SlateDBError::InvalidDBState);
+        };
+        Ok(Self {
+            manifest_store,
+            table_store,
+            options,
+            stored_manifest,
+            tokio_handle,
+            external_rx,
+            db_stats,
+        })
+    }
+
+    pub fn run(&mut self) {
```

Comment:
maybe related with the discussions around `block_on`, what do you think about making this `run()` method to be `async`? thus we could simply use the tokio primitives to schedule the tickers and reduce the `block_on` in a single place, in which simply being used to start up a new runtime.